File: algo/problems/0217-contains-duplicate/js/containsDuplicate.js
/**
 * 用 Set 边遍历边查，命中重复立即返回。O(n)/O(n)。
 * 一行写法：return new Set(nums).size !== nums.length（牺牲提前退出）。
 * @param {number[]} nums
 * @returns {boolean}
 */
function containsDuplicate(nums) {
  const seen = new Set();
  for (const x of nums) {
    if (seen.has(x)) return true;
    seen.add(x);
  }
  return false;
}

module.exports = { containsDuplicate };

File: algo/problems/0217-contains-duplicate/python/contains_duplicate.py
def contains_duplicate(nums: list[int]) -> bool:
    """集合去重：边遍历边查，命中重复立即返回。O(n)/O(n)。
    一行写法：return len(set(nums)) != len(nums)（牺牲提前退出）。"""
    seen: set[int] = set()
    for x in nums:
        if x in seen:
            return True
        seen.add(x)
    return False
